### DISCUSSION TITLE
complete details for a address map or memory map

### DISCUSSION THREAD

**Discussion by SikoVerilog** [SystemRDL/systemrdl-compiler#210] [Q&A] [answered]
Can you please tell how can I get starting address, range of address and number of registers in a address map in address map using python. I also need to know offset address of a register.

**Comment by amykyta3** [comment, marked as answer]
Please refer to the documentation. There are two comprehensive tutorials that will help you find what you need:
* https://systemrdl-compiler.readthedocs.io/en/stable/examples/print_hierarchy.html
* https://systemrdl-compiler.readthedocs.io/en/stable/examples/json_exporter.html
